desktop_version/src/DiscordNetwork.c

```diff
@@ -0,0 +1,165 @@
+
+
+struct DISCORD_application {
+    struct IDiscordCore* core;
+    struct IDiscordActivityManager* activityMan;
+} app;
+
+struct DiscordActivity activity;
+
+bool discordNotDetected = false;
+int discordCrashes = 0; // This is here to ensure we do not get stuck in a theoratical softlock of opening and crashing Discord instances.
+
+static void* libHandle = NULL;
+
+
+#define FUNC_LIST \
+    FOREACH_FUNC(enum EDiscordResult, DiscordCreate, (DiscordVersion, struct DiscordCreateParams*, struct IDiscordCore**))
+
+
+
+#define FOREACH_FUNC(rettype, name, params) static rettype (*name) params = NULL;
+FUNC_LIST
+#undef FOREACH_FUNC
+
+static void ClearPointers(void)
+{
+    SDL_UnloadObject(libHandle);
+    libHandle = NULL;
+#define FOREACH_FUNC(rettype, name, params) name = NULL;
+    FUNC_LIST
+#undef FOREACH_FUNC
+}
+
+void DISCORD_shutdown(void)
+{
+    if (libHandle)
+    {
+        ClearPointers();
+    }
+}
+
+int32_t DISCORD_init(void)
+{
+#if defined(__FreeBSD__) || defined(__OpenBSD__) || defined(__HAIKU__) || defined(__DragonFly__)
+    return 0;
+#endif
+    libHandle = SDL_LoadObject(DISCORD_LIBRARY);
+    if (!libHandle)
+    {
+        vlog_info(DISCORD_LIBRARY " not found!");
+        printf("Can't load object %s : %s\n", DISCORD_LIBRARY, SDL_GetError());
+        return 0;
+    }
+
+#define FOREACH_FUNC(rettype, name, params) \
+    name = (rettype (*) params) (intptr_t) SDL_LoadFunction(libHandle, #name); \
+    if (!name) \
+    { \
+        vlog_error(DISCORD_LIBRARY " symbol " #name " not found!"); \
+        ClearPointers(); \
+        return 0; \
+    }
+    FUNC_LIST
+#undef FOREACH_FUNC
+    memset(&app, 0, sizeof(app));
+
+    struct DiscordCreateParams params;
+    params.client_id = DISCORD_CLIENT_ID;
+    params.flags = DiscordCreateFlags_NoRequireDiscord;
+
+    if(!DiscordCreate(DISCORD_VERSION, &params, &app.core))
+    {
+        return 1;
+    }
+
+    if(app.core == NULL)
+    {
+	discordNotDetected = true;
+        return 0;
+    }
+
+
+    SDL_strlcpy(activity.assets.large_image, "vvvvvv", sizeof(activity.assets.large_image));
+    SDL_strlcpy(activity.assets.large_text, "Outside Dimension VVVVVV", sizeof(activity.assets.large_text));
+
+    app.activityMan = app.core->get_activity_manager(app.core);
+    app.activityMan->update_activity(app.activityMan, &activity, NULL, NULL);
+
+
+
+    return 0;
+}
+
+void DISCORD_REQUIRE(int x) {
+    if(discordNotDetected)
+    {
+        return;
+    }
+    if(discordCrashes > DISCORD_CRASH_LIMIT)
+    {
+        DISCORD_shutdown();
+        return;
+    }
+    if(x != DiscordResult_Ok)
+    {
+        ++discordCrashes;
+        DISCORD_shutdown();
+        DISCORD_init();
+        return;
+    }
+    discordCrashes = 0;
+}
+
+void DISCORD_update(const char *level, const char *name)
```

Comment:
@Daaaav GOGNetwork does not have the type* naming convention for functions I didn't implement, so I added that too. Should I just keep to my Discord implementation? Or would this fix be acceptable as part of this PR?

Reply:
I'd say it'd be acceptable to move spaces in the GOG/Steam code so that you don't have to add new code that either introduces `type *` convention or is inconsistent with the existing code. For more unrelated parts (functions you didn't implement for Discord but which were in the same GOG/Steam files you were already making consistency fixes in... Eh. Might as well.